XenModel/Actions/Network/ChangeNetworkingAction.cs

```diff
@@ -195,6 +201,13 @@ private void ReconfigureManagement(bool this_host, int hi)
                                                           clearDownManagementIP);
         }
 
+        private void PoolReconfigureManagement(int hi)
+        {
+            System.Diagnostics.Trace.Assert(newManagement != null);

```

Comment:
I suggest to have:
- an Assert for downManagement and
- an if for newManagement
E.g:
```
System.Diagnostics.Trace.Assert(downManagement != null);
if (newManagement == null) 
    return;
// the rest of the code here
```